plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/utils/ReplicationControllerResourceProcessor.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat - Initial Contribution
+ *******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.jboss.tools.openshift.common.core.connection.ConnectionsRegistry;
+import org.jboss.tools.openshift.core.connection.Connection;
+import com.openshift.restclient.OpenShiftException;
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.model.IReplicationController;
+import com.openshift.restclient.model.IResource;
+
+/**
+ * Processor for ReplicationControllerResourceProcessor. Scale ReplicationController to 0 and delete it.
+ * 
+ * @author Jeff Maury
+ *
+ */
+public class ReplicationControllerResourceProcessor extends BaseResourceProcessor {
+
+    public ReplicationControllerResourceProcessor() {
+        super(true);
+    }
+    
+    @Override
+    public void handleDelete(ConnectionsRegistry registry, Connection connection, IResource resource, boolean willDeleteSubResources)
+            throws OpenShiftException {
+        IReplicationController rc = (IReplicationController)resource;
+
+        try {
+            if (willDeleteSubResources) {
+                List<IResource>[] overlappingRCs = findOverlappingReplicationControllers(connection, rc);
+                if (!overlappingRCs[1].isEmpty()) {
+                    String list = overlappingRCs[1].stream().map(res -> res.getName()).collect(Collectors.joining(","));
+                    throw new OpenShiftException(
+                            "Found overlapping replication controllers for replication controller %s: %s", rc.getName(),
+                            list);
+                } else if (overlappingRCs[0].size() == 1) {
+                    rc.setDesiredReplicaCount(0);
+                    rc = connection.updateResource(rc);
+                    while (rc.getCurrentReplicaCount() > 0) {
+                        Thread.sleep(1000);
+                        rc = connection.getResource(rc);
```

Comment:
Is there a way for user to cancel it if something goes wrong and the expected result is not received for a long time?
And reporting to the progress monitor would be very welcome here.